modules/syslogformat/syslog-format.c

```diff
@@ -567,21 +568,12 @@ log_msg_parse_sd(LogMessage *self, const guchar **data, gint *length, const MsgF
           if (!isascii(*src) || *src == '=' || *src == ' ' || *src == ']' || *src == '"')
             goto error;
           /* read sd_id */
-          pos = 0;
+          g_string_truncate(sd_id_name, 0);
           while (left && *src != ' ' && *src != ']')
             {
-              /* the sd_id_name is max 32, the other chars are only stored in the self->sd_str*/
-              if (pos < sizeof(sd_id_name) - 1)
+              if (isascii(*src) && *src != '=' && *src != ' ' && *src != ']' && *src != '"')
                 {
-                  if (isascii(*src) && *src != '=' && *src != ' ' && *src != ']' && *src != '"')
-                    {
-                      sd_id_name[pos] = *src;
-                      pos++;
-                    }
-                  else
-                    {
-                      goto error;
-                    }
+                  g_string_append_c(sd_id_name, *src);
```

Comment:
with this change, we are removing any limits from the name of SDATA elements, which might be fine on its own, however we have a 255 limit on the length of name-value names

https://github.com/syslog-ng/syslog-ng/blob/04d68db1617b64a2458baee3caefe68df28830c1/lib/logmsg/nvtable.c#L65

Increasing that is a lot more difficult, as we are using a guint8 value to represent the length in NVTable (which is also serialized to disk).

So I would stick to static limits here, instead of using scratch buffers. Might also be good for performance.

I'd recommend a max size of 255 characters for each component and truncate the concatenated value at 255, this way we would still retain the name-value pair, albeit with a truncated name.